cupy/linalg/decomposition.py

```diff
@@ -371,22 +371,23 @@ def svd(a, full_matrices=True, compute_uv=True):
     if compute_uv:
         if full_matrices:
             u = cupy.empty((m, m), dtype=a_dtype)
-            vt = cupy.empty((n, n), dtype=a_dtype)
+            vt = x
```

Comment:
I expect we can use the slice view
```suggestion
            vt = x[:, :n]
```
, which keeps `vt.data.ptr == x.data.ptr`, to eliminate the slices in Lines 416, 418.

Reply:
Yes!